mne/viz/_brain/_brain.py

```diff
@@ -2310,6 +2318,89 @@ def add_label(self, label, color=None, alpha=1, scalar_thresh=None,
+        rois : list
+            The regions of interest to plot. See :func:`mne.get_montage_rois`
+            for one way to determine regions of interest. Regions can also be
+            chosen from :ref:`freesurfer lookup table`.
+        colors : list | matplotlib-style color | None
+            A list of anything matplotlib accepts: string, RGB, hex, etc.
+            (default Freesurfer lookup table colors).
+        alpha : float in [0, 1]
+            Alpha level to control opacity.
+        %(smooth)s
+        legend : bool | None
+            Add a legend displaying the names of the ``rois``. Default is
+            ``True`` if the number of ``rois`` is 10 or less.
+        **legend_kwargs : kwargs
+            Keyword arguments to be passed to
+            :meth:`pyvista.BasePlotter.add_legend`.
+
+        Notes
+        -----
+        .. versionadded:: 0.24
+        """
+        import nibabel as nib
+        from matplotlib.colors import colorConverter
+        from ...surface import marching_cubes
+
+        # load anatomical segmentation image
+        if not aseg.endswith('aseg'):
+            raise RuntimeError(
+                f'`aseg` file path must end with "aseg", got {aseg}')
+        aseg = _check_fname(op.join(self._subjects_dir, self._subject_id,
+                                    'mri', aseg + '.mgz'),
+                            overwrite='read', must_exist=True)
+        aseg = nib.load(aseg)
+        aseg_data = np.asarray(aseg.dataobj)
+        vox_mri_t = aseg.header.get_vox2ras_tkr()
+        mult = 1e-3 if self._units == 'm' else 1
+        vox_mri_t[:3] *= mult
+        del aseg
+
+        # read freesurfer lookup table
+        lut, fs_colors = read_freesurfer_lut()
+        if rois is None:  # assign default ROIs based on indices
+            lut_r = {v: k for k, v in lut.items()}
+            rois = [lut_r[idx] for idx in DEFAULTS['roi_indices']]
+
+        _validate_type(legend, (bool, None), 'legend')
+        if legend is None:
+            legend = len(rois) < 11
+
+        if colors is None:
+            colors = [fs_colors[roi] / 255 for roi in rois]
+        elif not isinstance(colors, (list, tuple)):
+            colors = [colors] * len(rois)  # make into list
+        colors = [colorConverter.to_rgba(color, alpha) for color in colors]
+        aseg_vals = set(np.unique(aseg_data))
+        for roi, color in zip(rois, colors):
+            val = lut[roi]
+            if val not in aseg_vals:
+                continue
+            mask = (aseg_data == val)
+            verts, triangles = marching_cubes(mask, level=0.5, smooth=smooth)
+            verts = apply_trans(vox_mri_t, verts)
+            self._renderer.mesh(
+                *verts.T, triangles=triangles, color=color, opacity=alpha,
+                reset_camera=False, render=False)
+
+        if legend:
+            self._renderer.plotter.add_legend(
+                list(zip(rois, colors)), **legend_kwargs)
```

Comment:
```suggestion
        if legend is not False:
            self._renderer.plotter.add_legend(
                list(zip(rois, colors)), **legend)
```

Reply:
It can't be `None`, it would have gotten reassigned earlier so I think zen is not to do this.

Reply:
It can't be None but it can be `dict()` and:
```
>>> bool(dict()) is False
True
```

Reply:
Oh sorry I was scrolling up from the bottom and I didn't see your refactoring comment above